### PR TITLE
global: initial configuration

### PR DESCRIPTION
**Pull request by mvesper**
Signed-off-by: Martin Vesper martin.vesper@cern.ch

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -34,7 +34,8 @@
 tests_require = [
     'check-manifest>=0.25',
     'coverage>=4.0',
-    'isort>=4.2.2',
+    'isort>=4.2.5',
```
Comment (jirikuncar) [inline setup.py]: can you keep the really minimal working requirements?

File: setup.py [hunk 2]
```diff
@@ -59,12 +60,11 @@
 ]
 
 install_requires = [
-    'Flask-BabelEx>=0.9.2',
+    'Flask-BabelEx>=0.9.3',
```
Comment (jirikuncar) [inline setup.py]: https://github.com/inveniosoftware/invenio-circulation/pull/8/files#r67860245

File: run-tests.sh [hunk 3]
```diff
@@ -27,6 +26,7 @@
 pydocstyle invenio_circulation && \
 isort -rc -c -df **/*.py && \
 check-manifest --ignore ".travis-*" && \
-sphinx-build -qnNW docs docs/_build/html && \
-python setup.py test && \
-sphinx-build -qnNW -b doctest docs docs/_build/doctest
+py.test tests/
```
Comment (jirikuncar) [inline run-tests.sh]: please revert changes in this file

File: tests/conftest.py [hunk 4]
```diff
@@ -25,17 +25,32 @@
 
 """Pytest configuration."""
 
-from __future__ import absolute_import, print_function
+from __future__ import absolute_import, print_function, unicode_literals
```
Comment (jirikuncar) [inline tests/conftest.py]: remove unicode_literals

File: invenio_circulation/ext.py [hunk 5]
```diff
@@ -43,6 +45,10 @@ def init_app(self, app):
     def init_config(self, app):
         """Initialize configuration."""
         app.config.setdefault(
-            'CIRCULATION_BASE_TEMPLATE',
-            app.config.get('BASE_TEMPLATE',
-                           'invenio_circulation/base.html'))
+            "CIRCULATION_BASE_TEMPLATE",
```
Comment (jirikuncar) [inline invenio_circulation/ext.py]: please use consistently single and double quotes

## 

(`s/"/'/` & `s/'''/"""/`)

File: invenio_circulation/config.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Invenio circulation configuration file."""
+
+CIRCULATION_EMAIL_SENDER = 'library.desk@cern.ch'
```
Comment (jirikuncar) [inline invenio_circulation/config.py]: - Please remove CERN specific email
- Can we have a default value `None` and warn user when (s)he tries to send an email or `before_first_app_request`?

File: invenio_circulation/config.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
```
Comment (jirikuncar) [inline invenio_circulation/config.py]: 2016

File: tests/test_invenio_circulation.py [hunk 8]
```diff
@@ -43,22 +45,25 @@ def test_version():
...
-    Babel(app)
-    InvenioCirculation(app)
-    app.register_blueprint(blueprint)
-    with app.test_client() as client:
-        res = client.get('/')
-        assert res.status_code == 200
-        assert 'Welcome to Invenio-Circulation' in str(res.data)
+    # check that current_circulation resolves correctly
+    with app.app_context():
+        current_circulation.init_app
```
Comment (jirikuncar) [inline tests/test_invenio_circulation.py]: `assert app.extensions['invenio-circulation'] == current_circulation._get_current_object()`

File: tests/test_invenio_circulation.py [hunk 9]
```diff
@@ -43,22 +45,25 @@ def test_version():
...
+    FlaskCLI(app)
+    app.url_map.converters['pid'] = PIDConverter
     ext = InvenioCirculation(app)
     assert 'invenio-circulation' in app.extensions
 
     app = Flask('testapp')
+    FlaskCLI(app)
+    app.url_map.converters['pid'] = PIDConverter
+
+    # check that current_circulation cannot be resolved
```
Comment (jirikuncar) [inline tests/test_invenio_circulation.py]: IMHO this is obvious from https://github.com/inveniosoftware/invenio-circulation/pull/8/files#diff-6500115c06a32cedd59b89d3aa4b3a98R63

File: MANIFEST.in [hunk 10]
```diff
@@ -22,21 +22,23 @@
...
 include LICENSE
 include babel.ini
 include pytest.ini
+include .dockerignore
+include .editorconfig
+include .tx/config
+recursive-include invenio_circulation *.py
+recursive-include invenio_circulation *.json
+recursive-include invenio_circulation *.msg
+recursive-include invenio_circulation *.html
```
Comment (egabancho) [inline MANIFEST.in]: maybe it's just me, but I would sort them

File: MANIFEST.in [hunk 11]
```diff
@@ -22,7 +22,7 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-
+include *.md
```
Comment (jirikuncar) [inline MANIFEST.in]: Do we really need it?

File: MANIFEST.in [hunk 12]
```diff
@@ -38,5 +38,7 @@ recursive-include docs *.rst
 recursive-include docs Makefile
 recursive-include examples *.py
 recursive-include invenio_circulation *.html
-recursive-include invenio_circulation *.po *.pot *.mo
```
Comment (jirikuncar) [inline MANIFEST.in:41]: please keep these as we should generate translations in next steps

File: run-tests.sh [hunk 13]
```diff
@@ -1,4 +1,3 @@
-#!/usr/bin/env sh
```
Comment (jirikuncar) [inline run-tests.sh:1]: We should keep it there

**Pull request merged**